src/main/ruby/core/posix.rb

```diff
@@ -217,6 +219,41 @@ def self.unsetenv(name)
   else
     raise 'Unsupported platform'
   end
+  
+  def self.write_string(fd, string, nonblock=false)
+    # TODO CS 14-Nov-17 this involves copying to a Java byte[], and then NFI will copy it again!
+    bytes = Truffle.invoke_primitive :string_to_null_terminated_byte_array, string
+    length = string.bytesize
+    to_write = length
+    while to_write > 0
+      ret = write(fd, bytes, to_write)
+      if ret < 0
+        if nonblock
+          errno = Errno.nfi_errno
+          if errno == Errno::EAGAIN::Errno || errno == Errno::EWOULDBLOCK::Errno
+            raise IO::EAGAINWaitWritable
+          end
+        end
+        Errno.handle_nfi
+      end
+      to_write -= ret
```

Comment:
If `write(2)` only writes part of the bytes (e.g. on a socket), then we should pass a buffer starting with new bytes.
This should be easy if we have a FFI::Pointer we can just pass (ptr + offset).